transfer_results_module/transfer_results.py

```diff
@@ -45,93 +38,114 @@ def get_results_from_firebase():
-            str(task_y), str(task_z), int(duplicates)]
-    # insert in table
-    m_con.query(sql_insert, data)
-    del m_con
+    for task_id, results in all_results.items():
+        for child_id, result in results.items():
+            key = 'results/{task_id}/{child_id}'.format(
+                task_id=task_id,
+                child_id=child_id)
 
-def data_transfer(q):
+            data[key] = None
+            #q.put([fb_db, task_id, child_id])
 
-    while not q.empty():
-        task_id, child_id, data = q.get()
+    fb_db.update(data)
+    print('finished deleting results')
+    logging.warning('deleted results in firebase')
+
+    del fb_db
 
-        try:
-            save_to_database(data)
-            delete_results_in_firebase(task_id, child_id)
-            #print('entry in mysql imported and deleted the following entry already.')
-            #print(task_id, child_id, data)
-            logging.info('entry in mysql imported and deleted the following entry already.')
-            logging.info('task_id: %s, child_id: %s' % (task_id, child_id))
-            q.task_done()
-
-        except pymysql.err.IntegrityError as e:
-            # we need to catch duplicate entries, 1062 --> "duplicate entry"
-            if e.args[0] == 1062:
-                print('Duplicate entry in mysql not imported. Will delete the following entry now.')
-                print(task_id, child_id, data)
-                logging.warning('Duplicate entry in mysql not imported. Will delete the following entry now.')
-                logging.warning('task_id: %s, child_id: %s' % (task_id, child_id))
-                delete_results_in_firebase(task_id, child_id)
-                q.task_done()
-        except:
-            print('error during data transfer. Do nothing.')
-            print(sys.exc_info())
-            logging.warning('error during data transfer. Do nothing.')
-            logging.warning(sys.exc_info())
-            q.task_done()
-
-def transfer_results(results_filename, all_results):
-    print('there are %s results to transfer.' % len(all_results))
-    logging.warning('there are %s results to transfer.' % len(all_results))
-
-
-    # we will use a queue to limit the number of threads running in parallel
-    q = Queue(maxsize=0)
-    num_threads = 8
 
+def results_to_txt(all_results):
+    results_txt_filename = 'raw_results.txt'
+    results_txt_file = open(results_txt_filename, 'w')
+
+    number_of_results = 0
     for task_id, results in all_results.items():
-        # there might be several results for a task_id
         for child_id, result in results.items():
-            # each result has a user_id and data
-            data = result['data']
-            q.put([task_id, child_id, data])
+            number_of_results += 1
+            outline = '{task_id}\t{user_id}\t{project_id}\t{timestamp}\t{result}\t{wkt}\t{task_x}\t{task_y}\t{task_z}\t{duplicates}\n'.format(
```

Comment:
It looks like you're wanting to write this out as tab-delimited content? I would tend to prefer using something like the [`csv`](https://docs.python.org/3.6/library/csv.html) module for stuff like this (you can specify TAB as the delimiter). In particular, the `DictWriter` is really convenient for stuff like this where you want to export all of the data like this. 

That said, this would probably be a little bit cleaner (and easier to maintain): 

```
outline = '\t'.join([
    task_id,
    result['data']['user'],
    result['data']['projectId'],
    result['data']['result'],
    result['data']['timestamp'],
    result['data']['wkt'],
    task_id.split('-')[1],
    task_id.split('-')[2],
    task_id.split('-')[0],
    0])
outline += '\n'

```

Note that this also requires that everything is a string. You could do this as well:

```
output_list = [
    task_id,
    result['data']['user'],
    result['data']['projectId'],
    result['data']['result'],
    result['data']['timestamp'],
    result['data']['wkt'],
    task_id.split('-')[1],
    task_id.split('-')[2],
    task_id.split('-')[0],
    0
]


outline = '\t'.join([str(s) for s in output_list])
outline += '\n'

```

Reply:
I agree that using the csv lib is cleaner and will make catching error easier. This way, you can be more certain that your output file will actually be usable.

Reply:
I'll :+1: @ericboucher's comment to my comment and add that I gave a workaround here but can't recommend more getting more familiar with the CSV module. If you want any pointers or examples or anything from us, please let us know

Reply:
Hey,
I adjusted this and now we use the csv module. I agree, that this is cleaner. :)